Update supervisor-and-application.markdown (#1618)
elixir-lang/elixir-lang.github.com: getting-started/mix-otp/supervisor-and-application.markdown

Although this chapter was the first time we implemented a supervisor, it was not the first time we used one! In the previous chapter, when we used `start_supervised!` to start the registry during our tests, `ExUnit` started the registry under a supervisor managed by the ExUnit framework itself. By defining our own supervisor, we provide more structure on how we initialize, shutdown and supervise processes in our applications, aligning our production code and tests with best practices.

> But we are not done yet. So far we are supervising the registry but our application is also starting buckets. Since buckets are started dynamically, we can also use a special type of supervisor called `DynamicSupervisor`, which provides some benefits that we will explore next.

Reviewer: ```suggestion
But we are not done yet. So far we are supervising the registry but our application is also starting buckets. Since buckets are started dynamically, we can use a special type of supervisor called `DynamicSupervisor`, which is optimized to handle such scenarios. Let's explore it next.
```
